tests: logging: dictionary: fix build error with MWDT toolchain and minimal libc

**Describe the bug**

Currently we get the build error when building `tests/subsys/logging/dictionary` with MWDT toolchain if the minimal libc library is used:

```

/verification_workspace/zephyr/tests/subsys/logging/dictionary/src/main.c:82:2: warning: implicit declaration of function 'fputc' is invalid in C99 [-Wimplicit-function-declaration]
        fputc('\r', stdout);
        ^
/verification_workspace/zephyr/tests/subsys/logging/dictionary/src/main.c:82:14: error: use of undeclared identifier 'stdout'
        fputc('\r', stdout);
                    ^
/verification_workspace/zephyr/tests/subsys/logging/dictionary/src/main.c:83:14: error: use of undeclared identifier 'stdout'
        fputc('\n', stdout);
                    ^
```

**Expected behavior**
No build error should happen

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: MWDT 2024.06
 - Commit SHA: recent main, 1ed04e7b49c5bec8e09f81480b7f95ae51b59b10

--- a/tests/subsys/logging/dictionary/src/main.c
+++ b/tests/subsys/logging/dictionary/src/main.c
@@ -8,6 +8,7 @@
 #include <inttypes.h>
 #include <zephyr/kernel.h>
 #include <string.h>
+#include <stdio.h>
 #include <zephyr/sys/printk.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/shell/shell.h>

